wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java

```diff
@@ -93,7 +94,8 @@ public OAuthToken getToken(String userId) throws IOException {
       if (token == null
           || token.getToken() == null
           || token.getToken().isEmpty()
-          || getJson(gitlabUserEndpoint, token.getToken(), GitLabUser.class) == null) {
+          || isNullOrEmpty(
+              getJson(gitlabUserEndpoint, token.getToken(), GitLabUser.class).getId())) {
         return null;
```

Comment:
returning null is a bad practice in general

Reply:
Agree, added a `TODO` for refactoring to `Optional`.